hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/upload/AbstractFileTransferHandler.java

```diff
@@ -94,18 +98,26 @@ protected VaadinMessageSource getI18n() {
     protected void startTransferToRepositoryThread(final InputStream inputStream, final FileUploadId fileUploadId,
             final String mimeType) {
         SpringContextHelper.getBean("asyncExecutor", ExecutorService.class).execute(
-                new TransferArtifactToRepositoryRunnable(inputStream, fileUploadId, mimeType, UI.getCurrent()));
+                new TransferArtifactToRepositoryRunnable(inputStream, fileUploadId, mimeType, UI.getCurrent(), uploadLock));
     }
 
     private void interruptUploadAndSetReason(final String failureReason) {
         uploadInterrupted = true;
         this.failureReason = failureReason;
     }
 
+    protected void interruptUploadDueToUploadFailed(final String msg) {
+        interruptUploadAndSetReason(StringUtils.isBlank(msg) ? i18n.getMessage("message.upload.failed") : msg);
```

Comment:
that is a bad idea to use plain string as msg/error reason, because it could be not localized and the user will see the system message

Reply:
Yes I can change that. What do you think of changing the generic value of `message.upload.failed` from `Streaming failed` to `Upload failed` I think this is easier to understand from user perspective then.

Reply:
Good idea!